src/test/java/itest/GraphQLIT.java

```diff
@@ -828,63 +801,51 @@ public void shouldReturnArchivedRecordingsFilteredByNames() throws Exception {
-                recordingInfo.getString("name"), Matchers.equalTo(TEST_RECORDING_NAME));
-        MatcherAssert.assertThat(recordingInfo.getString("state"), Matchers.equalTo("RUNNING"));
+        archiveRecordingFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
 
         Thread.sleep(2_000L);
-        for (int i = 0; i < 2; i++) {
-            // Archive the recording
-            CompletableFuture<Void> archiveRecordingFuture = new CompletableFuture<>();
-            webClient
-                    .patch(
-                            String.format(
-                                    "/api/v1/targets/%s/recordings/%s",
-                                    SELF_REFERENCE_TARGET_ID, TEST_RECORDING_NAME))
-                    .putHeader(HttpHeaders.CONTENT_TYPE.toString(), HttpMimeType.PLAINTEXT.mime())
-                    .sendBuffer(
-                            Buffer.buffer("SAVE"),
-                            ar -> {
-                                if (assertRequestStatus(ar, archiveRecordingFuture)) {
-                                    archiveRecordingFuture.complete(null);
-                                } else {
 
-                                    archiveRecordingFuture.completeExceptionally(
-                                            new RuntimeException("Archive request failed"));
-                                }
-                            });
+        // Retrieve the list of archived recordings
+        CompletableFuture<JsonArray> archivedRecordingsFuture = new CompletableFuture<>();
+        webClient
+                .get(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, archivedRecordingsFuture)) {
+                                archivedRecordingsFuture.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
 
-            archiveRecordingFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
-        }
+        JsonArray archivedRecordings =
+                archivedRecordingsFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
 
         // GraphQL Query to filter Archived recordings by names
         CompletableFuture<TargetNodesQueryResponse> resp2 = new CompletableFuture<>();
-        // List<String> filterNames = Arrays.asList("Recording1", "Recording2");
+
         String query =
                 "query { targetNodes {"
                         + "recordings {"
-                        + "archived(filter: { names: [\"io-cryostat-Cryostat_"
-                        + TEST_RECORDING_NAME
-                        + "_\\\\d{8}T\\\\d{6}Z(\\\\.\\\\d+)?\\\\.jfr\",\"io-cryostat-Cryostat_"
-                        + TEST_RECORDING_NAME
-                        + "_\\\\d{8}T\\\\d{6}Z(\\\\.\\\\d+)?\\\\.jfr\"] }) {"
+                        + "archived(filter: { names: [\"archivedRecording\",\"someOtherName\"] }) {"
```

Comment:
I don't think the archived file names will be like this, they would look like `io-cryostat-Cryostat_some-recording-name_12345678.jfr`, right?

I think the way to test this would be to use a different query, like `GET /api/v1/recordings`, to get the list of all of the archived recordings within the test. Then, within the test, you can apply some logic to filter out the ones that match your expectations, ie the `some-recording-name` matches the name you gave the recordings. Then you use that information to build the GraphQL query filter to pass in here and ensure that the GraphQL response also matches your expectation.

Reply:
yes the name expected should be the way you described

Reply:
I used ```GET /api/v1/recordings``` and then extracted the name of the archived recording to a variable1 then I used ```.post("/api/v2.2/graphql")``` and extraxted the archived recording to a variable2 then checked if they match.

Reply:
I'm not sure that this is resolved - the GraphQL query string here still asks the server for `"archivedRecording" and `"someOtherName"`, but those still shouldn't be the actual filenames of archived recordings. I thought that the solution would look like `archived(filter: { names: [" + retrievedArchivedRecordingsName + "] } )`

Reply:
this works too instead of hard cording the name